docs/user-guide/spark-user-guide.md

```diff
@@ -0,0 +1,159 @@
+
+# CQL Evaluation using Apache Spark
+
+## Overview
+
+The CQL evaluation function provided by the quality-measure-and-cohort-service project can be deployed in a variety of modes depending on the needs of the consuming solution. The most flexible option is a client-server mode of operation where the client performs the CQL evaluation for data that is stored in any remote FHIR server supporting the FHIR REST specification. While flexible, the client-server mode is likely not ideal for applications that need to process very large datasets due to the amount of data serialization and transfers required. For solutions wanting to process big data there is program provided that runs under Apache Spark. In this mode, the FHIR data model is eschewed in favor of traditional tabular data used by many analytics applications. Facilities are provided for loading the data from randomly ordered files, aggregated based on a specific execution context (e.g. Patient, Claim, etc.), and then evaluated using any number of CQL inputs. Outputs are written back out as tabular data. This guide describes how to configure and use that application.
+
+## Program assets
+
+The cohorting capabilities (aka [quality-measure-and-cohort-service](https://github.com/Alvearie/quality-measure-and-cohort-service/)) are delivered as a public open source project under the Alvearie organization in Github. The module of interest for users consuming the Apache Spark function is  [cohort-evaluator-spark](https://github.com/Alvearie/quality-measure-and-cohort-service/tree/cql-spark-evaluator). Provided inside that project is a sample [Dockerfile](https://github.com/Alvearie/quality-measure-and-cohort-service/blob/cql-spark-evaluator/cohort-evaluator-spark/Dockerfile) that can be built and published to the appropriate place in your execution environment. No publicly published version of the docker image exists at this time.
```

Comment:
Note for future reference - we'll need to update this paragraph once we have publicly available docker image of our cohort spark application available - which should be on our roadmap, pending toolchain function updates.

Reply:
Agree that we need a solution. I debated leaving this out entirely, but I tried to create the documentation with both internal and external users in mind, so I thought it good to at least make a statement. We can certainly revise later.